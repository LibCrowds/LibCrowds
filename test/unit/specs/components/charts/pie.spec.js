import BootstrapVue from 'bootstrap-vue'
import VueChartist from 'vue-chartist'
import 'chartist-plugin-tooltips'

import { mount, createLocalVue } from 'vue-test-utils'
import PieChart from '@/components/charts/Pie'

describe('Pie chart', () => {
  let localVue = null
  let wrapper = null
  let propsData = null
  let max = null
  let random = null

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(BootstrapVue)
    localVue.use(VueChartist)
    max = 100
    random = Math.floor(Math.random() * (max - 1 + 1)) + 1
    propsData = {
      header: 'My Pie Chart',
      chartData: {
        labels: ['Auth', 'Anon'],
        series: [
          { meta: 'Auth:', value: random },
          { meta: 'Anon', value: max - random }
        ]
      },
      unit: 'task'
    }
    wrapper = mount(PieChart, {
      localVue,
      propsData: propsData
    })
  })

  it('renders correctly', () => {
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('calculates the correct percentage for labels', () => {
    const idx = 0
    const label = propsData.chartData.labels[idx]
    const value = propsData.chartData.series[idx].value
    const expected = Math.round(value / max * 100) + '%'
    let modifiedLabel = wrapper.vm.options.labelInterpolationFnc(label, idx)
    expect(modifiedLabel).toBe(expected)
  })
})
