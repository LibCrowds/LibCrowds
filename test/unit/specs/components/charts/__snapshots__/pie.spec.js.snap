// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pie chart renders correctly 1`] = `

<div data-server-rendered="true"
     class="ct-container"
>
</div>

`;
